src/KeyEntityFormEnhancer.php

```diff
@@ -498,10 +509,20 @@ private function createSuggestion(\Exception $exception, KeyInterface $key): Mar
       // valid organization name and username provided with an invalid password
       // the MGMT server returns HTTP 500 with an error instead of HTTP 401.
       if ($exception->getCode() === 401 || ($exception->getCode() === 500 && $exception->getEdgeErrorCode() === 'usersandroles.SsoInternalServerError')) {
-        $suggestion = $this->t('@fail_text The given username (%username) or password is incorrect.', [
-          '@fail_text' => $fail_text,
-          '%username' => $key_type->getUsername($key),
-        ]);
+
+        // If on public cloud (using the default endpoint), the username should
+        // be an email.
+        if ($key_type->getEndpointType($key) == 'default' && !$this->emailValidator->isValid($key_type->getUsername($key))) {
```

Comment:
Use strict comparison

Reply:
@mxr576 can you explain why in this case? I am guessing you are thinking performance, but other than that I don't see an issue w/type juggling.

Reply:
Changed to strict comparison.